// @react-aria/menu/dist/module.js
import {useId as $d6bQb$useId, mergeProps as $d6bQb$mergeProps, filterDOMProps as $d6bQb$filterDOMProps, useSlotId as $d6bQb$useSlotId} from "@react-aria/utils";
import {useLongPress as $d6bQb$useLongPress, usePress as $d6bQb$usePress, useHover as $d6bQb$useHover, isFocusVisible as $d6bQb$isFocusVisible} from "@react-aria/interactions";
import {useMessageFormatter as $d6bQb$useMessageFormatter} from "@react-aria/i18n";
import {useOverlayTrigger as $d6bQb$useOverlayTrigger} from "@react-aria/overlays";
import {useSelectableList as $d6bQb$useSelectableList, useSelectableItem as $d6bQb$useSelectableItem} from "@react-aria/selection";
import {getItemCount as $d6bQb$getItemCount} from "@react-stately/collections";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $f00d7b16b5e8d6e2$exports = {};

$parcel$export($f00d7b16b5e8d6e2$exports, "useMenuTrigger", () => $f00d7b16b5e8d6e2$export$dc9c12ed27dd1b49);
var $f51f2c62e1c6fa57$exports = {};
var $adce0112009aa371$exports = {};
$adce0112009aa371$exports = JSON.parse("{\"longPressMessage\":\"اضغط مطولاً أو اضغط على Alt + السهم لأسفل لفتح القائمة\"}");


var $1d3f7d096c0dcb28$exports = {};
$1d3f7d096c0dcb28$exports = JSON.parse("{\"longPressMessage\":\"Натиснете продължително или натиснете Alt+ стрелка надолу, за да отворите менюто\"}");


var $875ee72102062854$exports = {};
$875ee72102062854$exports = JSON.parse("{\"longPressMessage\":\"Dlouhým stiskem nebo stisknutím kláves Alt + šipka dolů otevřete nabídku\"}");


var $0b027ea084fd351f$exports = {};
$0b027ea084fd351f$exports = JSON.parse("{\"longPressMessage\":\"Langt tryk eller tryk på Alt + pil ned for at åbne menuen\"}");


var $24e42099aa962232$exports = {};
$24e42099aa962232$exports = JSON.parse("{\"longPressMessage\":\"Drücken Sie lange oder drücken Sie Alt + Nach-unten, um das Menü zu öffnen\"}");


var $8b22544240ef2695$exports = {};
$8b22544240ef2695$exports = JSON.parse("{\"longPressMessage\":\"Πιέστε παρατεταμένα ή πατήστε Alt + κάτω βέλος για να ανοίξετε το μενού\"}");


var $1aba034f8273c2ea$exports = {};
$1aba034f8273c2ea$exports = JSON.parse("{\"longPressMessage\":\"Long press or press Alt + ArrowDown to open menu\"}");


var $1f635b4fc2c688b1$exports = {};
$1f635b4fc2c688b1$exports = JSON.parse("{\"longPressMessage\":\"Mantenga pulsado o pulse Alt + flecha abajo para abrir el menú\"}");


var $1e07d5f0de12cbdf$exports = {};
$1e07d5f0de12cbdf$exports = JSON.parse("{\"longPressMessage\":\"Menüü avamiseks vajutage pikalt või vajutage klahve Alt + allanool\"}");


var $139288a8fe39accd$exports = {};
$139288a8fe39accd$exports = JSON.parse("{\"longPressMessage\":\"Avaa valikko painamalla pohjassa tai näppäinyhdistelmällä Alt + Alanuoli\"}");


var $3de19cb0f92fa482$exports = {};
$3de19cb0f92fa482$exports = JSON.parse("{\"longPressMessage\":\"Appuyez de manière prolongée ou appuyez sur Alt + Flèche vers le bas pour ouvrir le menu.\"}");


var $76d1f987700e29e7$exports = {};
$76d1f987700e29e7$exports = JSON.parse("{\"longPressMessage\":\"לחץ לחיצה ארוכה או הקש Alt + ArrowDown כדי לפתוח את התפריט\"}");


var $6a724f202d89d6c5$exports = {};
$6a724f202d89d6c5$exports = JSON.parse("{\"longPressMessage\":\"Dugo pritisnite ili pritisnite Alt + strelicu prema dolje za otvaranje izbornika\"}");


var $39924d5fb7e5765c$exports = {};
$39924d5fb7e5765c$exports = JSON.parse("{\"longPressMessage\":\"Nyomja meg hosszan, vagy nyomja meg az Alt + lefele nyíl gombot a menü megnyitásához\"}");


var $48ca9160c777b181$exports = {};
$48ca9160c777b181$exports = JSON.parse("{\"longPressMessage\":\"Premere a lungo o premere Alt + Freccia giù per aprire il menu\"}");


var $2b2fe8258874b24d$exports = {};
$2b2fe8258874b24d$exports = JSON.parse("{\"longPressMessage\":\"長押しまたは Alt+下矢印キーでメニューを開く\"}");


var $7f3bcb0b01302e7d$exports = {};
$7f3bcb0b01302e7d$exports = JSON.parse("{\"longPressMessage\":\"길게 누르거나 Alt + 아래쪽 화살표를 눌러 메뉴 열기\"}");


var $717248addfa0b001$exports = {};
$717248addfa0b001$exports = JSON.parse("{\"longPressMessage\":\"Norėdami atidaryti meniu, nuspaudę palaikykite arba paspauskite „Alt + ArrowDown“.\"}");


var $250479c82d40d12f$exports = {};
$250479c82d40d12f$exports = JSON.parse("{\"longPressMessage\":\"Lai atvērtu izvēlni, turiet nospiestu vai nospiediet taustiņu kombināciju Alt + lejupvērstā bultiņa\"}");


var $b9ec90a40d5da860$exports = {};
$b9ec90a40d5da860$exports = JSON.parse("{\"longPressMessage\":\"Langt trykk eller trykk Alt + PilNed for å åpne menyen\"}");


var $84495c8c1264df78$exports = {};
$84495c8c1264df78$exports = JSON.parse("{\"longPressMessage\":\"Druk lang op Alt + pijl-omlaag of druk op Alt om het menu te openen\"}");


var $1ed1c4abe05d869a$exports = {};
$1ed1c4abe05d869a$exports = JSON.parse("{\"longPressMessage\":\"Naciśnij i przytrzymaj lub naciśnij klawisze Alt + Strzałka w dół, aby otworzyć menu\"}");


var $6c0bba061244d14f$exports = {};
$6c0bba061244d14f$exports = JSON.parse("{\"longPressMessage\":\"Pressione e segure ou pressione Alt + Seta para baixo para abrir o menu\"}");


var $f89477c065df8892$exports = {};
$f89477c065df8892$exports = JSON.parse("{\"longPressMessage\":\"Prima continuamente ou prima Alt + Seta Para Baixo para abrir o menu\"}");


var $8e1026548017f472$exports = {};
$8e1026548017f472$exports = JSON.parse("{\"longPressMessage\":\"Apăsați lung sau apăsați pe Alt + săgeată în jos pentru a deschide meniul\"}");


var $5ffa595127e3e13b$exports = {};
$5ffa595127e3e13b$exports = JSON.parse("{\"longPressMessage\":\"Нажмите и удерживайте или нажмите Alt + Стрелка вниз, чтобы открыть меню\"}");


var $c0307678aeb82973$exports = {};
$c0307678aeb82973$exports = JSON.parse("{\"longPressMessage\":\"Ponuku otvoríte dlhým stlačením alebo stlačením klávesu Alt + klávesu so šípkou nadol\"}");


var $00213556c76923e2$exports = {};
$00213556c76923e2$exports = JSON.parse("{\"longPressMessage\":\"Za odprtje menija pritisnite in držite gumb ali pritisnite Alt+puščica navzdol\"}");


var $cb21f39ff064ff18$exports = {};
$cb21f39ff064ff18$exports = JSON.parse("{\"longPressMessage\":\"Dugo pritisnite ili pritisnite Alt + strelicu prema dole da otvorite meni\"}");


var $f4f475fcd92bc63b$exports = {};
$f4f475fcd92bc63b$exports = JSON.parse("{\"longPressMessage\":\"Håll nedtryckt eller tryck på Alt + pil nedåt för att öppna menyn\"}");


var $599b86b59326de49$exports = {};
$599b86b59326de49$exports = JSON.parse("{\"longPressMessage\":\"Menüyü açmak için uzun basın veya Alt + Aşağı Ok tuşuna basın\"}");


var $dea30f1cbd6dd7fc$exports = {};
$dea30f1cbd6dd7fc$exports = JSON.parse("{\"longPressMessage\":\"Довго або звичайно натисніть комбінацію клавіш Alt і стрілка вниз, щоб відкрити меню\"}");


var $95a8ccc80c41dac8$exports = {};
$95a8ccc80c41dac8$exports = JSON.parse("{\"longPressMessage\":\"长按或按 Alt + 向下方向键以打开菜单\"}");


var $54a566c7316e6ef4$exports = {};
$54a566c7316e6ef4$exports = JSON.parse("{\"longPressMessage\":\"長按或按 Alt+向下鍵以開啟功能表\"}");


$f51f2c62e1c6fa57$exports = {
    "ar-AE": $adce0112009aa371$exports,
    "bg-BG": $1d3f7d096c0dcb28$exports,
    "cs-CZ": $875ee72102062854$exports,
    "da-DK": $0b027ea084fd351f$exports,
    "de-DE": $24e42099aa962232$exports,
    "el-GR": $8b22544240ef2695$exports,
    "en-US": $1aba034f8273c2ea$exports,
    "es-ES": $1f635b4fc2c688b1$exports,
    "et-EE": $1e07d5f0de12cbdf$exports,
    "fi-FI": $139288a8fe39accd$exports,
    "fr-FR": $3de19cb0f92fa482$exports,
    "he-IL": $76d1f987700e29e7$exports,
    "hr-HR": $6a724f202d89d6c5$exports,
    "hu-HU": $39924d5fb7e5765c$exports,
    "it-IT": $48ca9160c777b181$exports,
    "ja-JP": $2b2fe8258874b24d$exports,
    "ko-KR": $7f3bcb0b01302e7d$exports,
    "lt-LT": $717248addfa0b001$exports,
    "lv-LV": $250479c82d40d12f$exports,
    "nb-NO": $b9ec90a40d5da860$exports,
    "nl-NL": $84495c8c1264df78$exports,
    "pl-PL": $1ed1c4abe05d869a$exports,
    "pt-BR": $6c0bba061244d14f$exports,
    "pt-PT": $f89477c065df8892$exports,
    "ro-RO": $8e1026548017f472$exports,
    "ru-RU": $5ffa595127e3e13b$exports,
    "sk-SK": $c0307678aeb82973$exports,
    "sl-SI": $00213556c76923e2$exports,
    "sr-SP": $cb21f39ff064ff18$exports,
    "sv-SE": $f4f475fcd92bc63b$exports,
    "tr-TR": $599b86b59326de49$exports,
    "uk-UA": $dea30f1cbd6dd7fc$exports,
    "zh-CN": $95a8ccc80c41dac8$exports,
    "zh-TW": $54a566c7316e6ef4$exports
};






function $f00d7b16b5e8d6e2$export$dc9c12ed27dd1b49(props, state, ref) {
    let { type: type = 'menu' , isDisabled: isDisabled , trigger: trigger = 'press'  } = props;
    let menuTriggerId = $d6bQb$useId();
    let { triggerProps: triggerProps , overlayProps: overlayProps  } = $d6bQb$useOverlayTrigger({
        type: type
    }, state, ref);
    let onKeyDown = (e)=>{
        if (isDisabled) return;
        if (trigger === 'longPress' && !e.altKey) return;
        if (ref && ref.current) switch(e.key){
            case 'Enter':
            case ' ':
                if (trigger === 'longPress') return;
            // fallthrough
            case 'ArrowDown':
                // Stop propagation, unless it would already be handled by useKeyboard.
                if (!('continuePropagation' in e)) e.stopPropagation();
                e.preventDefault();
                state.toggle('first');
                break;
            case 'ArrowUp':
                if (!('continuePropagation' in e)) e.stopPropagation();
                e.preventDefault();
                state.toggle('last');
                break;
        }
    };
    let formatMessage = $d6bQb$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($f51f2c62e1c6fa57$exports)));
    let { longPressProps: longPressProps  } = $d6bQb$useLongPress({
        accessibilityDescription: formatMessage('longPressMessage'),
        onLongPressStart () {
            state.close();
        },
        onLongPress () {
            state.open('first');
        }
    });
    let pressProps = {
        onPressStart (e) {
            // For consistency with native, open the menu on mouse/key down, but touch up.
            if (e.pointerType !== 'touch' && e.pointerType !== 'keyboard' && !isDisabled) // If opened with a screen reader, auto focus the first item.
            // Otherwise, the menu itself will be focused.
            state.toggle(e.pointerType === 'virtual' ? 'first' : null);
        },
        onPress (e) {
            if (e.pointerType === 'touch' && !isDisabled) state.toggle();
        }
    };
    triggerProps = $d6bQb$mergeProps(triggerProps, trigger === 'press' ? pressProps : longPressProps);
    return {
        menuTriggerProps: {
            ...triggerProps,
            id: menuTriggerId,
            onKeyDown: onKeyDown
        },
        menuProps: {
            ...overlayProps,
            'aria-labelledby': menuTriggerId
        }
    };
}


var $c227569f61491ba9$exports = {};

$parcel$export($c227569f61491ba9$exports, "useMenu", () => $c227569f61491ba9$export$38eaa17faae8f579);


function $c227569f61491ba9$export$38eaa17faae8f579(props, state, ref) {
    let { shouldFocusWrap: shouldFocusWrap = true , ...otherProps } = props;
    if (!props['aria-label'] && !props['aria-labelledby']) console.warn('An aria-label or aria-labelledby prop is required for accessibility.');
    let domProps = $d6bQb$filterDOMProps(props, {
        labelable: true
    });
    let { listProps: listProps  } = $d6bQb$useSelectableList({
        ...otherProps,
        ref: ref,
        selectionManager: state.selectionManager,
        collection: state.collection,
        disabledKeys: state.disabledKeys,
        shouldFocusWrap: shouldFocusWrap
    });
    return {
        menuProps: $d6bQb$mergeProps(domProps, {
            role: 'menu',
            ...listProps
        })
    };
}


var $f9449620c639d150$exports = {};

$parcel$export($f9449620c639d150$exports, "useMenuItem", () => $f9449620c639d150$export$9d32628fc2aea7da);




function $f9449620c639d150$export$9d32628fc2aea7da(props, state, ref) {
    let { isSelected: isSelected , isDisabled: isDisabled , key: key , onClose: onClose , closeOnSelect: closeOnSelect , isVirtualized: isVirtualized , onAction: onAction  } = props;
    let role = 'menuitem';
    if (state.selectionManager.selectionMode === 'single') role = 'menuitemradio';
    else if (state.selectionManager.selectionMode === 'multiple') role = 'menuitemcheckbox';
    let labelId = $d6bQb$useSlotId();
    let descriptionId = $d6bQb$useSlotId();
    let keyboardId = $d6bQb$useSlotId();
    let ariaProps = {
        'aria-disabled': isDisabled,
        role: role,
        'aria-label': props['aria-label'],
        'aria-labelledby': labelId,
        'aria-describedby': [
            descriptionId,
            keyboardId
        ].filter(Boolean).join(' ') || undefined
    };
    if (state.selectionManager.selectionMode !== 'none') ariaProps['aria-checked'] = isSelected;
    if (isVirtualized) {
        ariaProps['aria-posinset'] = state.collection.getItem(key).index;
        ariaProps['aria-setsize'] = $d6bQb$getItemCount(state.collection);
    }
    let onKeyDown = (e)=>{
        // Ignore repeating events, which may have started on the menu trigger before moving
        // focus to the menu item. We want to wait for a second complete key press sequence.
        if (e.repeat) return;
        switch(e.key){
            case ' ':
                if (!isDisabled && state.selectionManager.selectionMode === 'none' && closeOnSelect !== false && onClose) onClose();
                break;
            case 'Enter':
                // The Enter key should always close on select, except if overridden.
                if (!isDisabled && closeOnSelect !== false && onClose) onClose();
                break;
        }
    };
    let onPressStart = (e)=>{
        if (e.pointerType === 'keyboard' && onAction) onAction(key);
    };
    let onPressUp = (e)=>{
        if (e.pointerType !== 'keyboard') {
            if (onAction) onAction(key);
            // Pressing a menu item should close by default in single selection mode but not multiple
            // selection mode, except if overridden by the closeOnSelect prop.
            if (onClose && (closeOnSelect ?? state.selectionManager.selectionMode !== 'multiple')) onClose();
        }
    };
    let { itemProps: itemProps  } = $d6bQb$useSelectableItem({
        selectionManager: state.selectionManager,
        key: key,
        ref: ref,
        shouldSelectOnPressUp: true
    });
    let { pressProps: pressProps  } = $d6bQb$usePress({
        onPressStart: onPressStart,
        onPressUp: onPressUp,
        isDisabled: isDisabled
    });
    let { hoverProps: hoverProps  } = $d6bQb$useHover({
        isDisabled: isDisabled,
        onHoverStart () {
            if (!$d6bQb$isFocusVisible()) {
                state.selectionManager.setFocused(true);
                state.selectionManager.setFocusedKey(key);
            }
        }
    });
    return {
        menuItemProps: {
            ...ariaProps,
            ...$d6bQb$mergeProps(itemProps, pressProps, hoverProps, {
                onKeyDown: onKeyDown
            })
        },
        labelProps: {
            id: labelId
        },
        descriptionProps: {
            id: descriptionId
        },
        keyboardShortcutProps: {
            id: keyboardId
        }
    };
}


var $eb7748cf1d7e6885$exports = {};

$parcel$export($eb7748cf1d7e6885$exports, "useMenuSection", () => $eb7748cf1d7e6885$export$73f7a44322579622);

function $eb7748cf1d7e6885$export$73f7a44322579622(props) {
    let { heading: heading , 'aria-label': ariaLabel  } = props;
    let headingId = $d6bQb$useId();
    return {
        itemProps: {
            role: 'presentation'
        },
        headingProps: heading ? {
            // Techincally, menus cannot contain headings according to ARIA.
            // We hide the heading from assistive technology, and only use it
            // as a label for the nested group.
            id: headingId,
            'aria-hidden': true
        } : {
        },
        groupProps: {
            role: 'group',
            'aria-label': ariaLabel,
            'aria-labelledby': heading ? headingId : undefined
        }
    };
}




export {$f00d7b16b5e8d6e2$export$dc9c12ed27dd1b49 as useMenuTrigger, $c227569f61491ba9$export$38eaa17faae8f579 as useMenu, $f9449620c639d150$export$9d32628fc2aea7da as useMenuItem, $eb7748cf1d7e6885$export$73f7a44322579622 as useMenuSection};
