// @react-stately/menu/dist/module.js
import {useOverlayTriggerState as $eMjaQ$useOverlayTriggerState} from "@react-stately/overlays";
import {useState as $eMjaQ$useState} from "react";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $a2e5b61c2187a058$exports = {};

$parcel$export($a2e5b61c2187a058$exports, "useMenuTriggerState", () => $a2e5b61c2187a058$export$79fefeb1c2091ac3);


function $a2e5b61c2187a058$export$79fefeb1c2091ac3(props) {
    let overlayTriggerState = $eMjaQ$useOverlayTriggerState(props);
    let [focusStrategy1, setFocusStrategy] = $eMjaQ$useState(null);
    return {
        focusStrategy: focusStrategy1,
        ...overlayTriggerState,
        open (focusStrategy = null) {
            setFocusStrategy(focusStrategy);
            overlayTriggerState.open();
        },
        toggle (focusStrategy = null) {
            setFocusStrategy(focusStrategy);
            overlayTriggerState.toggle();
        }
    };
}




export {$a2e5b61c2187a058$export$79fefeb1c2091ac3 as useMenuTriggerState};
