// @react-stately/radio/dist/module.js
import {useControlledState as $cKWHA$useControlledState} from "@react-stately/utils";
import {useMemo as $cKWHA$useMemo, useState as $cKWHA$useState} from "react";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $abdd678977242949$exports = {};

$parcel$export($abdd678977242949$exports, "useRadioGroupState", () => $abdd678977242949$export$bca9d026f8e704eb);


let $abdd678977242949$var$instance = Math.round(Math.random() * 10000000000);
let $abdd678977242949$var$i = 0;
function $abdd678977242949$export$bca9d026f8e704eb(props) {
    // Preserved here for backward compatibility. React Aria now generates the name instead of stately.
    let name = $cKWHA$useMemo(()=>props.name || `radio-group-${$abdd678977242949$var$instance}-${++$abdd678977242949$var$i}`
    , [
        props.name
    ]);
    let [selectedValue, setSelected] = $cKWHA$useControlledState(props.value, props.defaultValue, props.onChange);
    let [lastFocusedValue, setLastFocusedValue] = $cKWHA$useState(null);
    let setSelectedValue = (value)=>{
        if (!props.isReadOnly && !props.isDisabled) setSelected(value);
    };
    return {
        name: name,
        selectedValue: selectedValue,
        setSelectedValue: setSelectedValue,
        lastFocusedValue: lastFocusedValue,
        setLastFocusedValue: setLastFocusedValue,
        isDisabled: props.isDisabled || false,
        isReadOnly: props.isReadOnly || false
    };
}




export {$abdd678977242949$export$bca9d026f8e704eb as useRadioGroupState};
