// @react-stately/select/dist/module.js
import {useMenuTriggerState as $9aIFe$useMenuTriggerState} from "@react-stately/menu";
import {useSingleSelectListState as $9aIFe$useSingleSelectListState} from "@react-stately/list";
import {useState as $9aIFe$useState} from "react";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $643d032a500acdbf$exports = {};

$parcel$export($643d032a500acdbf$exports, "useSelectState", () => $643d032a500acdbf$export$5159ec8b34d4ec12);



function $643d032a500acdbf$export$5159ec8b34d4ec12(props) {
    let triggerState = $9aIFe$useMenuTriggerState(props);
    let listState = $9aIFe$useSingleSelectListState({
        ...props,
        onSelectionChange: (key)=>{
            if (props.onSelectionChange != null) props.onSelectionChange(key);
            triggerState.close();
        }
    });
    let [isFocused, setFocused] = $9aIFe$useState(false);
    return {
        ...listState,
        ...triggerState,
        open () {
            // Don't open if the collection is empty.
            if (listState.collection.size !== 0) triggerState.open();
        },
        toggle (focusStrategy) {
            if (listState.collection.size !== 0) triggerState.toggle(focusStrategy);
        },
        isFocused: isFocused,
        setFocused: setFocused
    };
}




export {$643d032a500acdbf$export$5159ec8b34d4ec12 as useSelectState};
