// @react-stately/slider/dist/module.js
import {snapValueToStep as $35Cqk$snapValueToStep, clamp as $35Cqk$clamp} from "@react-aria/utils";
import {useControlledState as $35Cqk$useControlledState} from "@react-stately/utils";
import {useState as $35Cqk$useState, useRef as $35Cqk$useRef} from "react";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $41e91674a959b70c$exports = {};

$parcel$export($41e91674a959b70c$exports, "useSliderState", () => $41e91674a959b70c$export$e5fda3247f5d67f9);



const $41e91674a959b70c$var$DEFAULT_MIN_VALUE = 0;
const $41e91674a959b70c$var$DEFAULT_MAX_VALUE = 100;
const $41e91674a959b70c$var$DEFAULT_STEP_VALUE = 1;
function $41e91674a959b70c$export$e5fda3247f5d67f9(props) {
    const { isDisabled: isDisabled , minValue: minValue = $41e91674a959b70c$var$DEFAULT_MIN_VALUE , maxValue: maxValue = $41e91674a959b70c$var$DEFAULT_MAX_VALUE , numberFormatter: formatter , step: step = $41e91674a959b70c$var$DEFAULT_STEP_VALUE  } = props;
    const [values, setValues] = $35Cqk$useControlledState(props.value, props.defaultValue ?? [
        minValue
    ], props.onChange);
    const [isDraggings, setDraggings] = $35Cqk$useState(new Array(values.length).fill(false));
    const isEditablesRef = $35Cqk$useRef(new Array(values.length).fill(true));
    const [focusedIndex, setFocusedIndex] = $35Cqk$useState(undefined);
    const valuesRef = $35Cqk$useRef(null);
    valuesRef.current = values;
    const isDraggingsRef = $35Cqk$useRef(null);
    isDraggingsRef.current = isDraggings;
    function getValuePercent(value) {
        return (value - minValue) / (maxValue - minValue);
    }
    function getThumbMinValue(index) {
        return index === 0 ? minValue : values[index - 1];
    }
    function getThumbMaxValue(index) {
        return index === values.length - 1 ? maxValue : values[index + 1];
    }
    function isThumbEditable(index) {
        return isEditablesRef.current[index];
    }
    function setThumbEditable(index, editable) {
        isEditablesRef.current[index] = editable;
    }
    function updateValue(index, value) {
        if (isDisabled || !isThumbEditable(index)) return;
        const thisMin = getThumbMinValue(index);
        const thisMax = getThumbMaxValue(index);
        // Round value to multiple of step, clamp value between min and max
        value = $35Cqk$snapValueToStep(value, thisMin, thisMax, step);
        valuesRef.current = $41e91674a959b70c$var$replaceIndex(valuesRef.current, index, value);
        setValues(valuesRef.current);
    }
    function updateDragging(index, dragging) {
        if (isDisabled || !isThumbEditable(index)) return;
        const wasDragging = isDraggingsRef.current[index];
        isDraggingsRef.current = $41e91674a959b70c$var$replaceIndex(isDraggingsRef.current, index, dragging);
        setDraggings(isDraggingsRef.current);
        // Call onChangeEnd if no handles are dragging.
        if (props.onChangeEnd && wasDragging && !isDraggingsRef.current.some(Boolean)) props.onChangeEnd(valuesRef.current);
    }
    function getFormattedValue(value) {
        return formatter.format(value);
    }
    function setThumbPercent(index, percent) {
        updateValue(index, getPercentValue(percent));
    }
    function getRoundedValue(value) {
        return Math.round((value - minValue) / step) * step + minValue;
    }
    function getPercentValue(percent) {
        const val = percent * (maxValue - minValue) + minValue;
        return $35Cqk$clamp(getRoundedValue(val), minValue, maxValue);
    }
    return {
        values: values,
        getThumbValue: (index)=>values[index]
        ,
        setThumbValue: updateValue,
        setThumbPercent: setThumbPercent,
        isThumbDragging: (index)=>isDraggings[index]
        ,
        setThumbDragging: updateDragging,
        focusedThumb: focusedIndex,
        setFocusedThumb: setFocusedIndex,
        getThumbPercent: (index)=>getValuePercent(values[index])
        ,
        getValuePercent: getValuePercent,
        getThumbValueLabel: (index)=>getFormattedValue(values[index])
        ,
        getFormattedValue: getFormattedValue,
        getThumbMinValue: getThumbMinValue,
        getThumbMaxValue: getThumbMaxValue,
        getPercentValue: getPercentValue,
        isThumbEditable: isThumbEditable,
        setThumbEditable: setThumbEditable,
        step: step
    };
}
function $41e91674a959b70c$var$replaceIndex(array, index, value) {
    if (array[index] === value) return array;
    return [
        ...array.slice(0, index),
        value,
        ...array.slice(index + 1)
    ];
}




export {$41e91674a959b70c$export$e5fda3247f5d67f9 as useSliderState};
