// @react-aria/toggle/dist/module.js
import {mergeProps as $aVyw8$mergeProps, filterDOMProps as $aVyw8$filterDOMProps} from "@react-aria/utils";
import {useFocusable as $aVyw8$useFocusable} from "@react-aria/focus";
import {usePress as $aVyw8$usePress} from "@react-aria/interactions";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $6f01a055f297c6b8$exports = {};

$parcel$export($6f01a055f297c6b8$exports, "useToggle", () => $6f01a055f297c6b8$export$cbe85ee05b554577);



function $6f01a055f297c6b8$export$cbe85ee05b554577(props, state, ref) {
    let { isDisabled: isDisabled = false , isRequired: isRequired , isReadOnly: isReadOnly , value: value , name: name , children: children , 'aria-label': ariaLabel , 'aria-labelledby': ariaLabelledby , validationState: validationState = 'valid'  } = props;
    let onChange = (e)=>{
        // since we spread props on label, onChange will end up there as well as in here.
        // so we have to stop propagation at the lowest level that we care about
        e.stopPropagation();
        state.setSelected(e.target.checked);
    };
    let hasChildren = children != null;
    let hasAriaLabel = ariaLabel != null || ariaLabelledby != null;
    if (!hasChildren && !hasAriaLabel) console.warn('If you do not provide children, you must specify an aria-label for accessibility');
    // This handles focusing the input on pointer down, which Safari does not do by default.
    let { pressProps: pressProps  } = $aVyw8$usePress({
        isDisabled: isDisabled
    });
    let { focusableProps: focusableProps  } = $aVyw8$useFocusable(props, ref);
    let interactions = $aVyw8$mergeProps(pressProps, focusableProps);
    let domProps = $aVyw8$filterDOMProps(props, {
        labelable: true
    });
    return {
        inputProps: $aVyw8$mergeProps(domProps, {
            'aria-invalid': validationState === 'invalid' || undefined,
            'aria-errormessage': props['aria-errormessage'],
            'aria-controls': props['aria-controls'],
            'aria-readonly': isReadOnly || undefined,
            'aria-required': isRequired || undefined,
            onChange: onChange,
            disabled: isDisabled,
            value: value,
            name: name,
            type: 'checkbox',
            ...interactions
        })
    };
}




export {$6f01a055f297c6b8$export$cbe85ee05b554577 as useToggle};
