// @react-aria/listbox/dist/main.js
var $cCiqH$reactariautils = require("@react-aria/utils");
var $cCiqH$reactariainteractions = require("@react-aria/interactions");
var $cCiqH$reactarialabel = require("@react-aria/label");
var $cCiqH$reactariaselection = require("@react-aria/selection");
var $cCiqH$reactstatelycollections = require("@react-stately/collections");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $791342f030d1fc2c$exports = {};

$parcel$export($791342f030d1fc2c$exports, "useListBox", () => $791342f030d1fc2c$export$50eacbbf140a3141);

var $55ba9445af8053ba$exports = {};

$parcel$export($55ba9445af8053ba$exports, "listData", () => $55ba9445af8053ba$export$3585ede4d035bf14);
$parcel$export($55ba9445af8053ba$exports, "getItemId", () => $55ba9445af8053ba$export$9145995848b05025);
const $55ba9445af8053ba$export$3585ede4d035bf14 = new WeakMap();
function $55ba9445af8053ba$var$normalizeKey(key) {
    if (typeof key === 'string') return key.replace(/\s*/g, '');
    return '' + key;
}
function $55ba9445af8053ba$export$9145995848b05025(state, itemKey) {
    let data = $55ba9445af8053ba$export$3585ede4d035bf14.get(state);
    if (!data) throw new Error('Unknown list');
    return `${data.id}-option-${$55ba9445af8053ba$var$normalizeKey(itemKey)}`;
}






function $791342f030d1fc2c$export$50eacbbf140a3141(props, state, ref) {
    let domProps = $cCiqH$reactariautils.filterDOMProps(props, {
        labelable: true
    });
    let { listProps: listProps  } = $cCiqH$reactariaselection.useSelectableList({
        ...props,
        ref: ref,
        selectionManager: state.selectionManager,
        collection: state.collection,
        disabledKeys: state.disabledKeys
    });
    let { focusWithinProps: focusWithinProps  } = $cCiqH$reactariainteractions.useFocusWithin({
        onFocusWithin: props.onFocus,
        onBlurWithin: props.onBlur,
        onFocusWithinChange: props.onFocusChange
    });
    // Share list id and some props with child options.
    let id = $cCiqH$reactariautils.useId(props.id);
    $55ba9445af8053ba$export$3585ede4d035bf14.set(state, {
        id: id,
        shouldUseVirtualFocus: props.shouldUseVirtualFocus,
        shouldSelectOnPressUp: props.shouldSelectOnPressUp,
        shouldFocusOnHover: props.shouldFocusOnHover,
        isVirtualized: props.isVirtualized
    });
    let { labelProps: labelProps , fieldProps: fieldProps  } = $cCiqH$reactarialabel.useLabel({
        ...props,
        id: id,
        // listbox is not an HTML input element so it
        // shouldn't be labeled by a <label> element.
        labelElementType: 'span'
    });
    return {
        labelProps: labelProps,
        listBoxProps: $cCiqH$reactariautils.mergeProps(domProps, focusWithinProps, state.selectionManager.selectionMode === 'multiple' ? {
            'aria-multiselectable': 'true'
        } : {
        }, {
            role: 'listbox',
            ...$cCiqH$reactariautils.mergeProps(fieldProps, listProps)
        })
    };
}


var $379f8fed3835a355$exports = {};

$parcel$export($379f8fed3835a355$exports, "useOption", () => $379f8fed3835a355$export$497855f14858aa34);





function $379f8fed3835a355$export$497855f14858aa34(props, state, ref) {
    let { key: key  } = props;
    let data = $55ba9445af8053ba$export$3585ede4d035bf14.get(state);
    let isDisabled = props.isDisabled ?? state.disabledKeys.has(key);
    let isSelected = props.isSelected ?? state.selectionManager.isSelected(key);
    let isFocused = state.selectionManager.focusedKey === key;
    let shouldSelectOnPressUp = props.shouldSelectOnPressUp ?? data.shouldSelectOnPressUp;
    let shouldFocusOnHover = props.shouldFocusOnHover ?? data.shouldFocusOnHover;
    let shouldUseVirtualFocus = props.shouldUseVirtualFocus ?? data.shouldUseVirtualFocus;
    let isVirtualized = props.isVirtualized ?? data.isVirtualized;
    let labelId = $cCiqH$reactariautils.useSlotId();
    let descriptionId = $cCiqH$reactariautils.useSlotId();
    let optionProps = {
        role: 'option',
        'aria-disabled': isDisabled,
        'aria-selected': state.selectionManager.selectionMode !== 'none' ? isSelected : undefined
    };
    // Safari with VoiceOver on macOS misreads options with aria-labelledby or aria-label as simply "text".
    // We should not map slots to the label and description on Safari and instead just have VoiceOver read the textContent.
    // https://bugs.webkit.org/show_bug.cgi?id=209279
    if (!($cCiqH$reactariautils.isMac() && $cCiqH$reactariautils.isWebKit())) {
        optionProps['aria-label'] = props['aria-label'];
        optionProps['aria-labelledby'] = labelId;
        optionProps['aria-describedby'] = descriptionId;
    }
    if (isVirtualized) {
        optionProps['aria-posinset'] = state.collection.getItem(key).index + 1;
        optionProps['aria-setsize'] = $cCiqH$reactstatelycollections.getItemCount(state.collection);
    }
    let { itemProps: itemProps , isPressed: isPressed  } = $cCiqH$reactariaselection.useSelectableItem({
        selectionManager: state.selectionManager,
        key: key,
        ref: ref,
        shouldSelectOnPressUp: shouldSelectOnPressUp,
        isVirtualized: isVirtualized,
        shouldUseVirtualFocus: shouldUseVirtualFocus,
        isDisabled: isDisabled
    });
    let { hoverProps: hoverProps  } = $cCiqH$reactariainteractions.useHover({
        isDisabled: isDisabled || !shouldFocusOnHover,
        onHoverStart () {
            if (!$cCiqH$reactariainteractions.isFocusVisible()) {
                state.selectionManager.setFocused(true);
                state.selectionManager.setFocusedKey(key);
            }
        }
    });
    return {
        optionProps: {
            ...optionProps,
            ...$cCiqH$reactariautils.mergeProps(itemProps, hoverProps),
            id: $55ba9445af8053ba$export$9145995848b05025(state, key)
        },
        labelProps: {
            id: labelId
        },
        descriptionProps: {
            id: descriptionId
        },
        isFocused: isFocused,
        isSelected: isSelected,
        isDisabled: isDisabled,
        isPressed: isPressed
    };
}


var $701722104b57f7b1$exports = {};

$parcel$export($701722104b57f7b1$exports, "useListBoxSection", () => $701722104b57f7b1$export$c3f9f39876e4bc7);

function $701722104b57f7b1$export$c3f9f39876e4bc7(props) {
    let { heading: heading , 'aria-label': ariaLabel  } = props;
    let headingId = $cCiqH$reactariautils.useId();
    return {
        itemProps: {
            role: 'presentation'
        },
        headingProps: heading ? {
            // Techincally, listbox cannot contain headings according to ARIA.
            // We hide the heading from assistive technology, and only use it
            // as a label for the nested group.
            id: headingId,
            'aria-hidden': true
        } : {
        },
        groupProps: {
            role: 'group',
            'aria-label': ariaLabel,
            'aria-labelledby': heading ? headingId : undefined
        }
    };
}



$parcel$exportWildcard(module.exports, $791342f030d1fc2c$exports);
$parcel$exportWildcard(module.exports, $379f8fed3835a355$exports);
$parcel$exportWildcard(module.exports, $701722104b57f7b1$exports);
$parcel$exportWildcard(module.exports, $55ba9445af8053ba$exports);
