// @react-aria/listbox/dist/module.js
import {filterDOMProps as $7Yq4N$filterDOMProps, useId as $7Yq4N$useId, mergeProps as $7Yq4N$mergeProps, useSlotId as $7Yq4N$useSlotId, isMac as $7Yq4N$isMac, isWebKit as $7Yq4N$isWebKit} from "@react-aria/utils";
import {useFocusWithin as $7Yq4N$useFocusWithin, useHover as $7Yq4N$useHover, isFocusVisible as $7Yq4N$isFocusVisible} from "@react-aria/interactions";
import {useLabel as $7Yq4N$useLabel} from "@react-aria/label";
import {useSelectableList as $7Yq4N$useSelectableList, useSelectableItem as $7Yq4N$useSelectableItem} from "@react-aria/selection";
import {getItemCount as $7Yq4N$getItemCount} from "@react-stately/collections";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $b6a6a3879de3c160$exports = {};

$parcel$export($b6a6a3879de3c160$exports, "useListBox", () => $b6a6a3879de3c160$export$50eacbbf140a3141);

var $d786742a15de2e64$exports = {};

$parcel$export($d786742a15de2e64$exports, "listData", () => $d786742a15de2e64$export$3585ede4d035bf14);
$parcel$export($d786742a15de2e64$exports, "getItemId", () => $d786742a15de2e64$export$9145995848b05025);
const $d786742a15de2e64$export$3585ede4d035bf14 = new WeakMap();
function $d786742a15de2e64$var$normalizeKey(key) {
    if (typeof key === 'string') return key.replace(/\s*/g, '');
    return '' + key;
}
function $d786742a15de2e64$export$9145995848b05025(state, itemKey) {
    let data = $d786742a15de2e64$export$3585ede4d035bf14.get(state);
    if (!data) throw new Error('Unknown list');
    return `${data.id}-option-${$d786742a15de2e64$var$normalizeKey(itemKey)}`;
}






function $b6a6a3879de3c160$export$50eacbbf140a3141(props, state, ref) {
    let domProps = $7Yq4N$filterDOMProps(props, {
        labelable: true
    });
    let { listProps: listProps  } = $7Yq4N$useSelectableList({
        ...props,
        ref: ref,
        selectionManager: state.selectionManager,
        collection: state.collection,
        disabledKeys: state.disabledKeys
    });
    let { focusWithinProps: focusWithinProps  } = $7Yq4N$useFocusWithin({
        onFocusWithin: props.onFocus,
        onBlurWithin: props.onBlur,
        onFocusWithinChange: props.onFocusChange
    });
    // Share list id and some props with child options.
    let id = $7Yq4N$useId(props.id);
    $d786742a15de2e64$export$3585ede4d035bf14.set(state, {
        id: id,
        shouldUseVirtualFocus: props.shouldUseVirtualFocus,
        shouldSelectOnPressUp: props.shouldSelectOnPressUp,
        shouldFocusOnHover: props.shouldFocusOnHover,
        isVirtualized: props.isVirtualized
    });
    let { labelProps: labelProps , fieldProps: fieldProps  } = $7Yq4N$useLabel({
        ...props,
        id: id,
        // listbox is not an HTML input element so it
        // shouldn't be labeled by a <label> element.
        labelElementType: 'span'
    });
    return {
        labelProps: labelProps,
        listBoxProps: $7Yq4N$mergeProps(domProps, focusWithinProps, state.selectionManager.selectionMode === 'multiple' ? {
            'aria-multiselectable': 'true'
        } : {
        }, {
            role: 'listbox',
            ...$7Yq4N$mergeProps(fieldProps, listProps)
        })
    };
}


var $ffbe2720d8d792ce$exports = {};

$parcel$export($ffbe2720d8d792ce$exports, "useOption", () => $ffbe2720d8d792ce$export$497855f14858aa34);





function $ffbe2720d8d792ce$export$497855f14858aa34(props, state, ref) {
    let { key: key  } = props;
    let data = $d786742a15de2e64$export$3585ede4d035bf14.get(state);
    let isDisabled = props.isDisabled ?? state.disabledKeys.has(key);
    let isSelected = props.isSelected ?? state.selectionManager.isSelected(key);
    let isFocused = state.selectionManager.focusedKey === key;
    let shouldSelectOnPressUp = props.shouldSelectOnPressUp ?? data.shouldSelectOnPressUp;
    let shouldFocusOnHover = props.shouldFocusOnHover ?? data.shouldFocusOnHover;
    let shouldUseVirtualFocus = props.shouldUseVirtualFocus ?? data.shouldUseVirtualFocus;
    let isVirtualized = props.isVirtualized ?? data.isVirtualized;
    let labelId = $7Yq4N$useSlotId();
    let descriptionId = $7Yq4N$useSlotId();
    let optionProps = {
        role: 'option',
        'aria-disabled': isDisabled,
        'aria-selected': state.selectionManager.selectionMode !== 'none' ? isSelected : undefined
    };
    // Safari with VoiceOver on macOS misreads options with aria-labelledby or aria-label as simply "text".
    // We should not map slots to the label and description on Safari and instead just have VoiceOver read the textContent.
    // https://bugs.webkit.org/show_bug.cgi?id=209279
    if (!($7Yq4N$isMac() && $7Yq4N$isWebKit())) {
        optionProps['aria-label'] = props['aria-label'];
        optionProps['aria-labelledby'] = labelId;
        optionProps['aria-describedby'] = descriptionId;
    }
    if (isVirtualized) {
        optionProps['aria-posinset'] = state.collection.getItem(key).index + 1;
        optionProps['aria-setsize'] = $7Yq4N$getItemCount(state.collection);
    }
    let { itemProps: itemProps , isPressed: isPressed  } = $7Yq4N$useSelectableItem({
        selectionManager: state.selectionManager,
        key: key,
        ref: ref,
        shouldSelectOnPressUp: shouldSelectOnPressUp,
        isVirtualized: isVirtualized,
        shouldUseVirtualFocus: shouldUseVirtualFocus,
        isDisabled: isDisabled
    });
    let { hoverProps: hoverProps  } = $7Yq4N$useHover({
        isDisabled: isDisabled || !shouldFocusOnHover,
        onHoverStart () {
            if (!$7Yq4N$isFocusVisible()) {
                state.selectionManager.setFocused(true);
                state.selectionManager.setFocusedKey(key);
            }
        }
    });
    return {
        optionProps: {
            ...optionProps,
            ...$7Yq4N$mergeProps(itemProps, hoverProps),
            id: $d786742a15de2e64$export$9145995848b05025(state, key)
        },
        labelProps: {
            id: labelId
        },
        descriptionProps: {
            id: descriptionId
        },
        isFocused: isFocused,
        isSelected: isSelected,
        isDisabled: isDisabled,
        isPressed: isPressed
    };
}


var $60eb9482f3936d5e$exports = {};

$parcel$export($60eb9482f3936d5e$exports, "useListBoxSection", () => $60eb9482f3936d5e$export$c3f9f39876e4bc7);

function $60eb9482f3936d5e$export$c3f9f39876e4bc7(props) {
    let { heading: heading , 'aria-label': ariaLabel  } = props;
    let headingId = $7Yq4N$useId();
    return {
        itemProps: {
            role: 'presentation'
        },
        headingProps: heading ? {
            // Techincally, listbox cannot contain headings according to ARIA.
            // We hide the heading from assistive technology, and only use it
            // as a label for the nested group.
            id: headingId,
            'aria-hidden': true
        } : {
        },
        groupProps: {
            role: 'group',
            'aria-label': ariaLabel,
            'aria-labelledby': heading ? headingId : undefined
        }
    };
}





export {$b6a6a3879de3c160$export$50eacbbf140a3141 as useListBox, $ffbe2720d8d792ce$export$497855f14858aa34 as useOption, $60eb9482f3936d5e$export$c3f9f39876e4bc7 as useListBoxSection, $d786742a15de2e64$export$3585ede4d035bf14 as listData, $d786742a15de2e64$export$9145995848b05025 as getItemId};
