// @react-aria/radio/dist/module.js
import {mergeProps as $iHgPh$mergeProps, filterDOMProps as $iHgPh$filterDOMProps, useId as $iHgPh$useId} from "@react-aria/utils";
import {useFocusable as $iHgPh$useFocusable, getFocusableTreeWalker as $iHgPh$getFocusableTreeWalker} from "@react-aria/focus";
import {usePress as $iHgPh$usePress, useFocusWithin as $iHgPh$useFocusWithin} from "@react-aria/interactions";
import {useLabel as $iHgPh$useLabel} from "@react-aria/label";
import {useLocale as $iHgPh$useLocale} from "@react-aria/i18n";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $6657631d22f179ad$exports = {};

$parcel$export($6657631d22f179ad$exports, "useRadio", () => $6657631d22f179ad$export$37b0961d2f4751e2);

const $bd4891a2aaf94c29$export$3b7b268d09480394 = new WeakMap();




function $6657631d22f179ad$export$37b0961d2f4751e2(props, state, ref) {
    let { value: value , children: children , 'aria-label': ariaLabel , 'aria-labelledby': ariaLabelledby  } = props;
    const isDisabled = props.isDisabled || state.isDisabled;
    let hasChildren = children != null;
    let hasAriaLabel = ariaLabel != null || ariaLabelledby != null;
    if (!hasChildren && !hasAriaLabel) console.warn('If you do not provide children, you must specify an aria-label for accessibility');
    let checked = state.selectedValue === value;
    let onChange = (e)=>{
        e.stopPropagation();
        state.setSelectedValue(value);
    };
    let { pressProps: pressProps  } = $iHgPh$usePress({
        isDisabled: isDisabled
    });
    let { focusableProps: focusableProps  } = $iHgPh$useFocusable($iHgPh$mergeProps(props, {
        onFocus: ()=>state.setLastFocusedValue(value)
    }), ref);
    let interactions = $iHgPh$mergeProps(pressProps, focusableProps);
    let domProps = $iHgPh$filterDOMProps(props, {
        labelable: true
    });
    let tabIndex = state.lastFocusedValue === value || state.lastFocusedValue == null ? 0 : -1;
    if (isDisabled) tabIndex = undefined;
    return {
        inputProps: $iHgPh$mergeProps(domProps, {
            ...interactions,
            type: 'radio',
            name: $bd4891a2aaf94c29$export$3b7b268d09480394.get(state),
            tabIndex: tabIndex,
            disabled: isDisabled,
            checked: checked,
            value: value,
            onChange: onChange
        })
    };
}


var $c6603e85cbfc3c71$exports = {};

$parcel$export($c6603e85cbfc3c71$exports, "useRadioGroup", () => $c6603e85cbfc3c71$export$62b9571f283ff5c2);






function $c6603e85cbfc3c71$export$62b9571f283ff5c2(props, state) {
    let { name: name , validationState: validationState , isReadOnly: isReadOnly , isRequired: isRequired , isDisabled: isDisabled , orientation: orientation = 'vertical'  } = props;
    let { direction: direction  } = $iHgPh$useLocale();
    let { labelProps: labelProps , fieldProps: fieldProps  } = $iHgPh$useLabel({
        ...props,
        // Radio group is not an HTML input element so it
        // shouldn't be labeled by a <label> element.
        labelElementType: 'span'
    });
    let domProps = $iHgPh$filterDOMProps(props, {
        labelable: true
    });
    // When the radio group loses focus, reset the focusable radio to null if
    // there is no selection. This allows tabbing into the group from either
    // direction to go to the first or last radio.
    let { focusWithinProps: focusWithinProps  } = $iHgPh$useFocusWithin({
        onBlurWithin () {
            if (!state.selectedValue) state.setLastFocusedValue(null);
        }
    });
    let onKeyDown = (e)=>{
        let nextDir;
        switch(e.key){
            case 'ArrowRight':
                if (direction === 'rtl' && orientation !== 'vertical') nextDir = 'prev';
                else nextDir = 'next';
                break;
            case 'ArrowLeft':
                if (direction === 'rtl' && orientation !== 'vertical') nextDir = 'next';
                else nextDir = 'prev';
                break;
            case 'ArrowDown':
                nextDir = 'next';
                break;
            case 'ArrowUp':
                nextDir = 'prev';
                break;
            default:
                return;
        }
        e.preventDefault();
        let walker = $iHgPh$getFocusableTreeWalker(e.currentTarget, {
            from: e.target
        });
        let nextElem;
        if (nextDir === 'next') {
            nextElem = walker.nextNode();
            if (!nextElem) {
                walker.currentNode = e.currentTarget;
                nextElem = walker.firstChild();
            }
        } else {
            nextElem = walker.previousNode();
            if (!nextElem) {
                walker.currentNode = e.currentTarget;
                nextElem = walker.lastChild();
            }
        }
        if (nextElem) {
            // Call focus on nextElem so that keyboard navigation scrolls the radio into view
            nextElem.focus();
            state.setSelectedValue(nextElem.value);
        }
    };
    let groupName = $iHgPh$useId(name);
    $bd4891a2aaf94c29$export$3b7b268d09480394.set(state, groupName);
    return {
        radioGroupProps: $iHgPh$mergeProps(domProps, {
            // https://www.w3.org/TR/wai-aria-1.2/#radiogroup
            role: 'radiogroup',
            onKeyDown: onKeyDown,
            'aria-invalid': validationState === 'invalid' || undefined,
            'aria-errormessage': props['aria-errormessage'],
            'aria-readonly': isReadOnly || undefined,
            'aria-required': isRequired || undefined,
            'aria-disabled': isDisabled || undefined,
            'aria-orientation': orientation,
            ...fieldProps,
            ...focusWithinProps
        }),
        labelProps: labelProps
    };
}




export {$6657631d22f179ad$export$37b0961d2f4751e2 as useRadio, $c6603e85cbfc3c71$export$62b9571f283ff5c2 as useRadioGroup};
