// @react-aria/live-announcer/dist/module.js
import $cqdCi$react, {useState as $cqdCi$useState, useImperativeHandle as $cqdCi$useImperativeHandle, Fragment as $cqdCi$Fragment} from "react";
import $cqdCi$reactdom from "react-dom";
import {VisuallyHidden as $cqdCi$VisuallyHidden} from "@react-aria/visually-hidden";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $b95a471aac7c72bc$exports = {};

$parcel$export($b95a471aac7c72bc$exports, "announce", () => $b95a471aac7c72bc$export$a9b970dcc4ae71a9);
$parcel$export($b95a471aac7c72bc$exports, "clearAnnouncer", () => $b95a471aac7c72bc$export$d10ae4f68404609a);
$parcel$export($b95a471aac7c72bc$exports, "destroyAnnouncer", () => $b95a471aac7c72bc$export$d8686216b8b81b2f);



/* Inspired by https://github.com/AlmeroSteyn/react-aria-live */ const $b95a471aac7c72bc$var$LIVEREGION_TIMEOUT_DELAY = 7000;
let $b95a471aac7c72bc$var$liveRegionAnnouncer = /*#__PURE__*/ $cqdCi$react.createRef();
let $b95a471aac7c72bc$var$node = null;
let $b95a471aac7c72bc$var$messageId = 0;
function $b95a471aac7c72bc$export$a9b970dcc4ae71a9(message, assertiveness = 'assertive', timeout = $b95a471aac7c72bc$var$LIVEREGION_TIMEOUT_DELAY) {
    $b95a471aac7c72bc$var$ensureInstance((announcer)=>announcer.announce(message, assertiveness, timeout)
    );
}
function $b95a471aac7c72bc$export$d10ae4f68404609a(assertiveness) {
    $b95a471aac7c72bc$var$ensureInstance((announcer)=>announcer.clear(assertiveness)
    );
}
function $b95a471aac7c72bc$export$d8686216b8b81b2f() {
    if ($b95a471aac7c72bc$var$liveRegionAnnouncer.current) {
        $cqdCi$reactdom.unmountComponentAtNode($b95a471aac7c72bc$var$node);
        document.body.removeChild($b95a471aac7c72bc$var$node);
        $b95a471aac7c72bc$var$node = null;
    }
}
/**
 * Ensures we only have one instance of the announcer so that we don't have elements competing.
 */ function $b95a471aac7c72bc$var$ensureInstance(callback) {
    if (!$b95a471aac7c72bc$var$liveRegionAnnouncer.current) {
        $b95a471aac7c72bc$var$node = document.createElement('div');
        $b95a471aac7c72bc$var$node.dataset.liveAnnouncer = 'true';
        document.body.prepend($b95a471aac7c72bc$var$node);
        $cqdCi$reactdom.render(/*#__PURE__*/ $cqdCi$react.createElement($b95a471aac7c72bc$var$LiveRegionAnnouncer, {
            ref: $b95a471aac7c72bc$var$liveRegionAnnouncer
        }), $b95a471aac7c72bc$var$node, ()=>callback($b95a471aac7c72bc$var$liveRegionAnnouncer.current)
        );
    } else callback($b95a471aac7c72bc$var$liveRegionAnnouncer.current);
}
const $b95a471aac7c72bc$var$LiveRegionAnnouncer = /*#__PURE__*/ $cqdCi$react.forwardRef((_, ref)=>{
    let [assertiveMessages, setAssertiveMessages] = $cqdCi$useState([]);
    let [politeMessages, setPoliteMessages] = $cqdCi$useState([]);
    let clear = (assertiveness)=>{
        if (!assertiveness || assertiveness === 'assertive') setAssertiveMessages([]);
        if (!assertiveness || assertiveness === 'polite') setPoliteMessages([]);
    };
    let $b95a471aac7c72bc$export$a9b970dcc4ae71a9 = (message1, assertiveness = 'assertive', timeout = $b95a471aac7c72bc$var$LIVEREGION_TIMEOUT_DELAY)=>{
        let id = $b95a471aac7c72bc$var$messageId++;
        if (assertiveness === 'assertive') setAssertiveMessages((messages)=>[
                ...messages,
                {
                    id: id,
                    text: message1
                }
            ]
        );
        else setPoliteMessages((messages)=>[
                ...messages,
                {
                    id: id,
                    text: message1
                }
            ]
        );
        if (message1 !== '') setTimeout(()=>{
            if (assertiveness === 'assertive') setAssertiveMessages((messages)=>messages.filter((message)=>message.id !== id
                )
            );
            else setPoliteMessages((messages)=>messages.filter((message)=>message.id !== id
                )
            );
        }, timeout);
    };
    $cqdCi$useImperativeHandle(ref, ()=>({
            announce: $b95a471aac7c72bc$export$a9b970dcc4ae71a9,
            clear: clear
        })
    );
    return(/*#__PURE__*/ $cqdCi$react.createElement($cqdCi$Fragment, null, /*#__PURE__*/ $cqdCi$react.createElement($b95a471aac7c72bc$var$MessageBlock, {
        "aria-live": "assertive"
    }, assertiveMessages.map((message)=>/*#__PURE__*/ $cqdCi$react.createElement("div", {
            key: message.id
        }, message.text)
    )), /*#__PURE__*/ $cqdCi$react.createElement($b95a471aac7c72bc$var$MessageBlock, {
        "aria-live": "polite"
    }, politeMessages.map((message)=>/*#__PURE__*/ $cqdCi$react.createElement("div", {
            key: message.id
        }, message.text)
    ))));
});
function $b95a471aac7c72bc$var$MessageBlock({ children: children , 'aria-live': ariaLive  }) {
    return(/*#__PURE__*/ $cqdCi$react.createElement($cqdCi$VisuallyHidden, {
        role: "log",
        "aria-live": ariaLive,
        "aria-relevant": "additions"
    }, children));
}




export {$b95a471aac7c72bc$export$a9b970dcc4ae71a9 as announce, $b95a471aac7c72bc$export$d10ae4f68404609a as clearAnnouncer, $b95a471aac7c72bc$export$d8686216b8b81b2f as destroyAnnouncer};
